discovery-kubernetes-api/src/main/scala/akka/discovery/kubernetes/KubernetesApiSimpleServiceDiscovery.scala

```diff
@@ -99,16 +103,28 @@ class KubernetesApiSimpleServiceDiscovery(system: ActorSystem) extends SimpleSer
       entity <- response.entity.toStrict(resolveTimeout)
 
       podList <- {
-        log.debug("Kubernetes API entity: [{}]", entity.data.utf8String)
-
-        val unmarshalled = Unmarshal(entity).to[PodList]
 
-        unmarshalled.failed.foreach { t =>
-          log.error(t, "Failed to unmarshal Kubernetes API response status [{}]; check RBAC settings",
-            response.status.value)
+        response.status match {
+          case StatusCodes.OK =>
+            log.debug("Kubernetes API entity: [{}]", entity.data.utf8String)
+            val unmarshalled = Unmarshal(entity).to[PodList]
+            unmarshalled.failed.foreach { t =>
+              log.error(t, "Failed to unmarshal Kubernetes API response.  Status code: [{}]; Response body: [{}]",
```

Comment:
:+1:

Reply:
Maybe 'error' is a bit strong? At least 'warning' though indeed

Reply:
I'll switch to warning. The only case now is a version mis-match so nothing is likely to work.